File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxi = 0;
        int n = height.length;
        int l=0, r = n-1;
        while(l<r){
            maxi = Math.max(maxi, Math.min(height[l],height[r])*(r-l));
            int tmp = (height[l]<height[r]) ? l++: r--;
        }
        return maxi;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height)-1
        ans = 0
        while l<r:
            area = (r-l)*(min(height[l], height[r]))
            if height[l]< height[r]:
                l += 1
            else:
                r -= 1
            
            ans = max(ans, area)
        
        return ans
            